crypto/keys/sr25519/sr25519.go

```diff
@@ -0,0 +1,151 @@
+package sr25519
+
+import (
+	"bytes"
+	"crypto/subtle"
+	"fmt"
+
+	"github.com/tendermint/tendermint/crypto"
+	tmsr25519 "github.com/tendermint/tendermint/crypto/sr25519"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var _ cryptotypes.PrivKey = &PrivKey{}
+var _ codec.AminoMarshaler = &PrivKey{}
+
+const (
+	PubKeySize  = 32
+	PrivKeySize = 32
+	keyType     = "sr25519"
+	PrivKeyName = "tendermint/PrivKeySr25519"
+	PubKeyName  = "tendermint/PubKeySr25519"
+)
+
+// Bytes returns the byte representation of the Private Key.
+func (privKey *PrivKey) Bytes() []byte {
+	return privKey.MiniSecretKey
+}
+
+// Sign produces a signature on the provided message.
+func (privKey *PrivKey) Sign(msg []byte) ([]byte, error) {
+	return tmsr25519.PrivKey(privKey.MiniSecretKey).Sign(msg)
+}
+
+// PubKey gets the corresponding public key from the private key.
+func (privKey *PrivKey) PubKey() cryptotypes.PubKey {
+	return &PubKey{Key: tmsr25519.PrivKey(privKey.MiniSecretKey).PubKey().Bytes()}
+}
+
+// Equals - you probably don't need to use this.
+// Runs in constant time based on length of the
+func (privKey *PrivKey) Equals(other cryptotypes.LedgerPrivKey) bool {
+	return privKey.Type() == other.Type() && subtle.ConstantTimeCompare(privKey.Bytes(), other.Bytes()) == 1
```

Comment:
```suggestion
	return privKey.Type() == other.Type() && subtle.ConstantTimeCompare(privKey.MiniSecretKey, other.MiniSecretKey) == 1
```

Reply:
I'm not sure if `ConstantTimeCompare` vs `bytes.Equal` will do any difference here.